File: src/main/java/me/saro/kit/webs/WebIgnoreCertificate.java
package me.saro.kit.webs;

import javax.net.ssl.HttpsURLConnection;
import javax.net.ssl.SSLContext;
import javax.net.ssl.TrustManager;
import javax.net.ssl.X509TrustManager;
import java.net.HttpURLConnection;
import java.security.KeyManagementException;
import java.security.NoSuchAlgorithmException;
import java.security.cert.X509Certificate;

/**
 * Web Ignore Certificate
 * @author PARK Yong Seo
 * @since 1.0.0
 */
public class WebIgnoreCertificate {
    private WebIgnoreCertificate() {
    }

    final static TrustManager[] ALL_TRUST_MANAGER = new TrustManager[] {
            new X509TrustManager() {
                @Override
                public X509Certificate[] getAcceptedIssuers() {
                    return null;
                }

                @Override
                public void checkClientTrusted(X509Certificate[] certs, String authType) {
                }

                @Override
                public void checkServerTrusted(X509Certificate[] certs, String authType) {
                }
            }
    };

    /**
     * casing ignore certificate to HttpURLConnection
     * @param connection
     * @throws KeyManagementException
     * @throws NoSuchAlgorithmException
     */
    public static void ignoreCertificate(HttpURLConnection connection) throws KeyManagementException, NoSuchAlgorithmException {
        SSLContext sslContext = SSLContext.getInstance("SSL");
        sslContext.init(null, ALL_TRUST_MANAGER, new java.security.SecureRandom());
        ((HttpsURLConnection)connection).setSSLSocketFactory(sslContext.getSocketFactory());
        ((HttpsURLConnection)connection).setHostnameVerifier((hostname, session) -> true);
    }

    /**
     * apply global ignore certificate
     * @throws KeyManagementException
     * @throws NoSuchAlgorithmException
     */
    public static void applyGlobalIgnoreCertificate() throws KeyManagementException, NoSuchAlgorithmException {
        SSLContext sslContext = SSLContext.getInstance("SSL");
        sslContext.init(null, ALL_TRUST_MANAGER, new java.security.SecureRandom());
        HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.getSocketFactory());
        HttpsURLConnection.setDefaultHostnameVerifier((hostname, session) -> true);
    }
}

File: src/test/java/me/saro/test/legacy/webs/WebIgnoreCertificate.kt
package me.saro.test.legacy.webs

import java.net.HttpURLConnection
import java.security.KeyManagementException
import java.security.NoSuchAlgorithmException
import java.security.SecureRandom
import java.security.cert.X509Certificate
import javax.net.ssl.*

/**
 * Web Ignore Certificate
 * @author PARK Yong Seo
 * @since 1.0.0
 */
object WebIgnoreCertificate {
    private val ALL_TRUST_MANAGER = arrayOf<TrustManager>(
        object : X509TrustManager {
            override fun getAcceptedIssuers(): Array<X509Certificate> = arrayOf()
            override fun checkClientTrusted(certs: Array<X509Certificate>, authType: String) {}
            override fun checkServerTrusted(certs: Array<X509Certificate>, authType: String) {}
        }
    )

    /**
     * casing ignore certificate to HttpURLConnection
     * @param connection
     * @throws KeyManagementException
     * @throws NoSuchAlgorithmException
     */
    @JvmStatic
    @Throws(KeyManagementException::class, NoSuchAlgorithmException::class)
    fun ignoreCertificate(connection: HttpURLConnection) {
        val sslContext = SSLContext.getInstance("SSL")
        sslContext.init(null, ALL_TRUST_MANAGER, SecureRandom())
        (connection as HttpsURLConnection).sslSocketFactory = sslContext.socketFactory
        connection.hostnameVerifier = HostnameVerifier { hostname: String?, session: SSLSession? -> true }
    }

    /**
     * apply global ignore certificate
     * @throws KeyManagementException
     * @throws NoSuchAlgorithmException
     */
    @Throws(KeyManagementException::class, NoSuchAlgorithmException::class)
    fun applyGlobalIgnoreCertificate() {
        val sslContext = SSLContext.getInstance("SSL")
        sslContext.init(null, ALL_TRUST_MANAGER, SecureRandom())
        HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.socketFactory)
        HttpsURLConnection.setDefaultHostnameVerifier { hostname: String?, session: SSLSession? -> true }
    }
}